intermediate_notebooks/E2E/taxi/NYCTaxi-E2E-GPU-Dask.ipynb

```diff
@@ -0,0 +1,1995 @@
+       "2      1    1            4           0   714          40.739998   \n",
+       "3      1    1            4           0   672          40.790001   \n",
+       "4      1    1            4           0   666          40.719997   \n",
+       "\n",
+       "   pickup_longitude_r  dropoff_latitude_r  dropoff_longitude_r  h_distance  \n",
+       "0          -73.989998           40.759998           -73.919998    6.040600  \n",
+       "1          -74.000000           40.779999           -73.970001    8.113065  \n",
+       "2          -73.979996           40.750000           -74.000000    1.484953  \n",
+       "3          -73.949997           40.860001           -73.930000    8.579659  \n",
+       "4          -74.000000           40.680000           -74.000000    3.952547  "
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def haversine_dist(df):\n",
+    "    h_distance = cuspatial.haversine_distance(df['pickup_longitude'], df['pickup_latitude'], df['dropoff_longitude'], df['dropoff_latitude'])\n",
+    "    df['h_distance']= h_distance\n",
+    "    df['h_distance']= df['h_distance'].astype('float32')\n",
+    "    return df\n",
+    "\n",
+    "taxi_df = taxi_df.map_partitions(haversine_dist)\n",
+    "taxi_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Pick a Training Set\n",
+    "\n",
+    "Let's imagine you're making a trip to New York on the 25th and want to build a model to predict what fare prices will be like the last few days of the month based on the first part of the month. We'll use a query expression to identify the `day` of the month to use to divide the data into train and test sets.\n",
+    "\n",
+    "The wall-time below represents how long it takes your GPU cluster to load data from the Google Cloud Storage bucket and the ETL portion of the workflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#since we calculated the h_distance let's drop the trip_distance column, and then do model training with XGB.\n",
+    "taxi_df = taxi_df.drop('trip_distance', axis=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "7884075\n"
+     ]
+    }
+   ],
+   "source": [
+    "# this is the original data partition for train and test sets.\n",
+    "X_train = taxi_df.query('day < 25').persist()\n",
+    "\n",
+    "# create a Y_train ddf with just the target variable\n",
+    "Y_train = X_train[['fare_amount']].persist()\n",
+    "# drop the target variable from the training ddf\n",
+    "X_train = X_train[X_train.columns.difference(['fare_amount'])]\n",
+    "\n",
+    "# this wont return until all data is in GPU memory\n",
+    "done = wait([X_train, Y_train])\n",
```

Comment:
I get the same `ratecodeid` error here:
```
ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   ['RatecodeID']
  Missing: ['ratecodeid']
```

Reply:
Strange, I was not getting such error.. will check it out.

Reply:
@randerzander I do not get the error below. I used `yellow_tripdata_2016-01.csv`.
```ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   ['RatecodeID']
  Missing: ['ratecodeid']
```

I have now calculating RMSE issue again between two dask_cudf.core.Series.